src/main/java/mat/client/shared/CQLLibraryResultTable.java

```diff
@@ -62,41 +64,75 @@ public Boolean getValue(CQLLibraryDataSetObject object) {
 			public SafeHtml getValue(CQLLibraryDataSetObject object) {
 				return getCQLLibraryNameColumnToolTip(object);
 			}
-		};
-//		Double Click event on Grid row to navigate to Composer Tab.
-		if(MatContext.get().getMatOnFHIR().getFlagOn()) {
-			table.addDomHandler(new DoubleClickHandler() {
-				@Override
-				public void onDoubleClick(DoubleClickEvent event) {
-					for (CQLLibraryDataSetObject object : selectionModel.getSelectedSet()) {
-						SelectionEvent.fire(fireEvent, object);
-					}
+			/*
+				Single Click to select row and enable checkbox
+				Double Click to navigate to CQL Composer tab
+			*/
+			@Override
+			public void onBrowserEvent(Cell.Context context, Element elem, CQLLibraryDataSetObject object, NativeEvent event) {
+				if (lastClick < System.currentTimeMillis() - 500)
```

Comment:
Doesn't comply with DRY https://en.wikipedia.org/wiki/Don%27t_repeat_yourself
Same logic repeated several times

Reply:
Event handlers over the row isn't working as expected, I have tried multiple ways to do it over the row. Though it's not the optimal way of doing it, it does its job.

Reply:
In any case you don't have to repeat same code several times